### ISSUE TITLE
Incorrect page param behavior

### ISSUE THREAD

**Issue by flood4life** [intercom/intercom-ruby#269]
I've replied with this in #215, but looks like my comment went unnoticed.

API documentation states that `page` param defines what page of result set to fetch as first, not as the only one. Perhaps it's better to distinguish these two params? E.g.,  `intercom.users.find_all(order: :asc, start_page: 2, end_page: 10)` would retrieve users from 51 to 500, while `intercom.users.find_all(order: :asc, start_page: 2, end_page: 2)` would retrieve only second page of users.

## Version info
  - intercom-ruby version: 3.5.2

## Expected behavior
`intercom.users.find_all(order: :asc, page: 2)` returns all users except for first page (users 51..last)
## Actual behavior
`intercom.users.find_all(order: :asc, page: 2)` returns only seconds page of users (users 51..100)

**Comment by choran** [comment]
@flood4life sorry I missed that comment in the other issue.
Can you point out where in the API docs it states that? The behavior in the API itself is the same as the ruby SDK, e.g. url -u APP_ID:APIKEY -H 'Accept: application/json' 'https://api.intercom.io/users?per_page=5&page=4' will return the 4th page with five results, not all the results from page 4 on. So maybe it is confusing somewhere in the doc? I can clarify it to make it clearer.

**Comment by flood4life** [comment]
@choran 

Request: `curl -u APP_ID:API_KEY -H 'Accept: application/json' 'https://api.intercom.io/users?per_page=5&page=4'`
Response: `{"type":"user.list","pages":{"type":"pages","next":"https://api.intercom.io/users?per_page=5&page=5","page":4,"per_page":5,"total_pages":83}, "users":[.....]}`

Request: `curl -u APP_ID:API_KEY -H 'Accept: application/json' 'https://api.intercom.io/users?per_page=5'` #page omitted
Response: `{"type":"user.list","pages":{"type":"pages","next":"https://api.intercom.io/users?per_page=5&page=2","page":1,"per_page":5,"total_pages":83}, "users":[.....]}`

API behaves identically with and without `page` param in first request, i.e., it allows to iterate through pages after initial response.

https://developers.intercom.io/reference#list-users
page -> what page of results to fetch *defaults to first page*.
The *defaults to first page* implies that `?per_page=5` and `?per_page=5&page=1` behave identically.

**Comment by choran** [comment]
@flood4life 
yeah, so we try and ensure that the ruby SDK matches the behavior of the API itself. (Let me know if I have misunderstood you here but I think that is what you are referring to in your last post?)
Is this clearer now or is there still something I can help clarify?

**Comment by flood4life** [comment]
@choran 
Nope, that's not the point.
API never responds with the WHOLE set of users, right? Instead it responds with a page of users and a link to the next page.
Responses to both requests (with `page` and without) contain the `pages` object, which contains `next` url, which is used to retrieve next page. 
`intercom.users.find_all(per_page: 50)` sends `https://api.intercom.io/users?per_page=50`, receives first page with 50 users, then uses `pages.next` to get second page, etc. This ultimately iterates over all users.
`intercom.users.find_all(per_page: 50, page: 1)` sends `https://api.intercom.io/users?per_page=50&page=1`, and its response does not differ from previous example at all.

Thus, the only difference in API responses is the start page of the result. It's still possible to iterate over pages after the start page. That's why I do not understand why Ruby SDK should stop at single page.

It was possible to achieve same behavior in client code: 

```ruby
intercom.users.find_all(page: page, per_page: per_page).each_slice(per_page) do |slice|
    slice.map{ |u| do_work_on_user(u) }
    break
end
```

Now it's not possible to over all users starting from certain page. And there are use case for this: multi-threaded processing. E.g., each thread processes 50 pages, and the `page` param (pre 3.5.2) is perfect for this.

**Comment by choran** [comment]
@flood4life 
Aha, I see, interesting, thanks for clarifying. 
Unfortunately, the way it previously worked was not by design. As noted in [#215](https://github.com/intercom/intercom-ruby/issues/215), there was an issue with the way pagination was working and in response the that issue we updated it. At the moment we do not plan on changing this. 

But I think it should still be possible to help with your use-case.
For example, you could call the uses endpoint without the find_all wrapper if you want to replicate the API and have access to customize it to suit you specific needs. e.g.
```
pry(main)> users = intercom.get("/users", per_page:5, page:3);   
pry(main)> users['pages']
=> {"type"=>"pages",
    "next"=>"https://api.intercom.io/users?per_page=5&page=4",
    "page"=>3,
    "per_page"=>5,
    "total_pages"=>18}
pry(main)> users['users'][0]['name']
=> "Robert Richardson"
pry(main)> users['users'][3]['name']
=> "Helen Wallace"
```
You could then have access to the next page and iterate through the pages you like starting at a different page. You can choose how to set it up in ways that suit you particular needs. I am not 100% sure if this meets you use-case so let me know if this helps?

**Comment by flood4life** [comment]
@choran 
I already have code that depends on previous behavior of `ClientCollectionProxy`, and it's more DRY (I have to iterate not only over users) and readable. So, if you're not planning to at least allow old behavior via option, I'll have to stick to my fork.

**Comment by choran** [comment]
Unfortunately, at the moment we are not planning on changing the functionality we implemented in #215. Thanks for the clarification and feedback on this issue.
